src/gsy_e/models/strategy/energy_parameters/energy_params_eb.py

```diff
@@ -21,37 +21,407 @@
+        reference_slot = market_slot.set(hour=12, minute=0)
+
+        if state.get_energy_production_forecast_kWh(
+                reference_slot) <= FLOATING_POINT_TOLERANCE:
+            scaling_factor = 0.
+        else:
+            scaling_factor = (
+                    state.get_available_energy_kWh(reference_slot) /
+                    state.get_energy_production_forecast_kWh(reference_slot))
+        return abs(scaling_factor) * peak_energy_kWh
+
+    def event_load_traded_energy(
+            self, energy_kWh: float, market_slot: pendulum.DateTime, state: "LoadState"):
+        trade_profile = self._profile_generator.generate_trade_profile(
+            energy_kWh=energy_kWh,
+            market_slot=market_slot,
+            product_type=self._product_type)
+
+        for time_slot, energy_value_kWh in trade_profile.items():
+            state.decrement_energy_requirement(
+                purchased_energy_Wh=energy_value_kWh * 1000,
+                time_slot=time_slot,
+                area_name=self._area.name)
+
+    def event_pv_traded_energy(
+            self, energy_kWh: float, market_slot: pendulum.DateTime, state: "PVState"):
+        # Create a new profile that spreads the trade energy across multiple slots. The values
+        # of this new profile are obtained by scaling the values of the standard solar profile
+        trade_profile = self._profile_generator.generate_trade_profile(
+            energy_kWh=energy_kWh,
+            market_slot=market_slot,
+            product_type=self._product_type)
+
+        for time_slot, energy_value_kWh in trade_profile.items():
+            state.decrement_available_energy(
+                sold_energy_kWh=energy_value_kWh,
+                time_slot=time_slot,
+                area_name=self._area.name)
+
+
+class ForwardEnergyParams(ABC):
+    """Common abstract base class for the energy parameters of the forward strategies."""
+    def __init__(self):
+        self._posted_energy_kWh: DefaultDict = defaultdict(lambda: 0.)
+        self._forward_energy_params = {
+            AvailableMarketTypes.INTRADAY: _IntradayEnergyParams(self._posted_energy_kWh),
+            AvailableMarketTypes.DAY_FORWARD: _DayForwardEnergyParams(self._posted_energy_kWh),
+            AvailableMarketTypes.WEEK_FORWARD: _LongForwardEnergyParameters(
+                self._posted_energy_kWh, AvailableMarketTypes.WEEK_FORWARD),
+            AvailableMarketTypes.MONTH_FORWARD: _LongForwardEnergyParameters(
+                self._posted_energy_kWh, AvailableMarketTypes.MONTH_FORWARD),
+            AvailableMarketTypes.YEAR_FORWARD: _LongForwardEnergyParameters(
+                self._posted_energy_kWh, AvailableMarketTypes.YEAR_FORWARD)
+        }
+
+        self._area = None
+        self._profile_generator: Optional[ForwardTradeProfileGenerator] = None
+
+    def get_posted_energy_kWh(
+            self, market_slot: pendulum.DateTime, market_type: AvailableMarketTypes) -> float:
```

Comment:
Although I do not like the name, we should be consistent, I think [it was decided](https://github.com/gridsingularity/gsy-e/pull/1534#discussion_r979841354) to use `product_type` from now on right?

Reply:
Let me fix this here too, thanks!